src/shared/components/cancerSummary/CancerSummary.tsx

```diff
@@ -0,0 +1,448 @@
+    @observable private yAxis: 'alt-freq' | 'abs-count' = 'alt-freq';
+    @observable private xAxis: 'y-axis' | 'can-types' = 'y-axis';
+    @observable private multiSelectValue:string = this.cancerTypes[0].value;
+    @observable private multiSelect:any = {
+        options: this.cancerTypes,
+        value: this.multiSelectValue
+    };
+    @observable private altCasesValue = 0;
+    @observable private totalCasesValue = 0;
+
+    constructor(props:ICancerSummaryContentProps) {
+        super(props);
+
+        this.handleYAxisChange = this.handleYAxisChange.bind(this);
+        this.handleXAxisChange = this.handleXAxisChange.bind(this);
+        this.handleAltSliderChange = this.handleAltSliderChange.bind(this);
+        this.handleGenomicCheckboxChange = this.handleGenomicCheckboxChange.bind(this);
+        this.handleSelectChange = this.handleSelectChange.bind(this);
+        this.handleTotalSliderChange = this.handleTotalSliderChange.bind(this);
+        this.toggleShowControls = this.toggleShowControls.bind(this);
+        this.setPngAnchor = this.setPngAnchor.bind(this);
+        this.setPdfAnchor = this.setPdfAnchor.bind(this);
+        this.downloadPdf = this.downloadPdf.bind(this);
+        this.resetSliders = this.resetSliders.bind(this);
+    }
+
+    @computed get datasets() {
+        const {data} = this.props;
+        const {yAxis, showGenomicAlt, multiSelectValue:msv} = this;
+        return _.without(
+            _.map(data, (cancer:ICancerAlterationData, cancerTypeS:string) => {
+                const totalCases = cancer.total;
+                const cancerAlterations:ICancerAlterationPlotData = _.omit(cancer, ['total']);
+                const altTotalCount = _.reduce(cancerAlterations, (total:number, value:number) => total + value, 0);
+                const altTotalPercent = this.toRoundedNumber(altTotalCount / totalCases * 100);
+                if (_.intersection(["all", "", cancerTypeS.toLowerCase()], msv.split(",").map(val => val.toLowerCase())).length && totalCases >= this.totalCasesValue) {
+                    return {
+                        label: cancerTypeS,
+                        sortBy: yAxis,
+                        symbol: yAxis === "abs-count" ? '' : "%",
+                        sortCount: yAxis === "abs-count" ? altTotalCount : altTotalPercent,
+                        data: _.map(cancerAlterations, (count:number, altType: string) => {
+                            const percent = count / totalCases * 100;
+                            const total = yAxis === "abs-count" ? count : percent;
+                            return {
+                                label: altType,
+                                totalCases,
+                                altTotalPercent,
+                                total,
+                                count,
+                                percent,
+                                backgroundColor: showGenomicAlt  ? this.getColors(altType) : '#aaaaaa'
```

Comment:
lets get the hex values into some configuration map

Reply:
I'm not sure I know what configuration map means: Isn't what I have on line 189?  I just did the same thing as https://github.com/cBioPortal/cbioportal-frontend/blob/842b4d69f5e9747661e475b696bf2dd4cbcdf299/src/shared/components/barGraph/BarGraph.tsx#L33

Reply:
yes, 189 is right approach, but the #aaaa should also be a constant or in a map.